Add financial assistance page skeleton and Zendesk submission.

peter-fogg opened the pull request:
# [ECOM-3038](https://openedx.atlassian.net/browse/ECOM-3038)

@AlasdairSwan Here's the skeleton for the financial assistance page. Right now you can see that the view is passing the user and a list of enrolled courses [here](https://github.com/edx/edx-platform/blob/peter-fogg/fa-page/lms/djangoapps/courseware/views.py#L1425-L1427) -- I'm not sure what, if any, other information you'll need on the JS side of things. Let's talk tomorrow and I'll make whatever changes are necessary.

@bderusha, can you review as well?

singingwolfboy commented on lms/templates/financial-assistance/apply.html:
```diff
@@ -0,0 +1,52 @@
+    <ol class="apply-form-list">
+      <li class="apply-form-section">
+        <h2>${_("About Me")}</h2>
+        <p>${_("The following information is part of your edX profile. All fields are required. If any information is incorrect, {link_start}edit your profile{link_end} before submitting this application.").format(
```
use `settings.PLATFORM_NAME` instead of "edX"

peter-fogg replied:
Thanks for catching that. It's worth noting, though, that this is an edx-only page which is configured off by default for OpenEdX instances.

bderusha commented:
:+1:

peter-fogg commented:
@bderusha This PR has been expanded to include the Zendesk submissions coming from the Backbone form that @AlasdairSwan is working on. Can you take a second look when you get a chance?

bderusha commented on lms/djangoapps/courseware/tests/test_views.py:
```diff
@@ -458,6 +460,135 @@ def _email_opt_in_checkbox(self, response, org_name_string=None):
+    def test_submit_financial_assistance_request(self, mock_record_feedback):
+        username = self.user.username
+        course = unicode(self.course_key)
+        legal_name = 'Jesse Pinkman'
```
+1

bderusha commented on lms/djangoapps/courseware/tests/test_views.py:
```diff
@@ -458,6 +460,135 @@ def _email_opt_in_checkbox(self, response, org_name_string=None):
+        self.assertEqual(response.status_code, 400)
+
+    def test_submit_financial_assistance_wrong_username(self):
+        response = self._submit_financial_assistance_form({'username': 'wwhite'})
```
Might be good to use ddt here for this method and the one above. Not sure if we would also want to add other cases using ddt

peter-fogg replied:
Good idea, I'll update this.

bderusha commented on lms/djangoapps/courseware/views.py:
```diff
@@ -1404,3 +1409,229 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True):
+    ' audit track for a course that offers Verified Certificates, and then complete this application.'
+    ' Note that you must complete a separate application for each course you take.'
+).format(
+    percent_sign="%",
```
Why is this passed vs just putting it in the string?

peter-fogg replied:
`ugettext` tries to do its own printf-style formatting, which breaks when a single percent sign is included. Passing the percent sign in seems like the easiest workaround.

bderusha replied:
Can we just escape it with `%%`?  Not a deal breaker, but I think it's cleaner to escape than to include it as a named variable given that it will never change.

peter-fogg replied:
Yeah, that was the first thing I tried -- unfortunately that results two percent signs shown in the UI.

bderusha commented:
Part of me feels weird about how much text is in the view, but I don't know how else I would do it to share the text with backbone. So yea, other than my few minor comments this looks good

peter-fogg commented:
@bderusha I've addressed your comment and removed the default "Choose One" option from the view. I agree that it's not great to have so much text in the view, but it seems like this is the best way to do it since so much needs to be shared with Backbone.

bderusha commented:
One NIT.  Given the timing, up to you about the fix.
:+1:
